knowhere/index/vector_index/IndexDiskANN.cpp

```diff
@@ -319,6 +435,10 @@ IndexDiskANN<T>::QueryByRange(const DatasetPtr& dataset_ptr, const Config& confi
         *(p_lims + row + 1) = res_count;
     }
 
+    if (ex.has_value()) {
+        throw ex.value();
```

Comment:
why not use KNOWHERE_THROW_MSG ?

Reply:
ex is `optional<KnowhereException>` which we get from `KNOWHERER_THROW_MSG`. Since we cannot throw exceptions from openmp threads, so we collect them and do rethrow here